### PR TITLE
Add draw button

### PR DESCRIPTION
**Pull request by benshumaker**
Closes: #57
Screen Shot 2022-12-16 at 10 28 19 AM
Screen Shot 2022-12-16 at 10 28 04 AM

### CODE REVIEW

File: frontend/packages/web/src/components/features/DrawButton/DrawButton.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  const [player, _] = useRecoilState(playerState);
+  const [game, setGame] = useRecoilState(gameState);
+
+  const context = React.useContext(ServicesContext);
+  const homepageService = useMemo(
+    () => new HomepageService(context.roomHttpService, context.roomWebSocketService),
+    [context.roomHttpService, context.roomWebSocketService]
+  );
+
+  const handleDrawClick = (accepted: boolean) => {
```
Comment (izakfr) [inline frontend/packages/web/src/components/features/DrawButton/DrawButton.tsx:18]: Can we change this to `handleDrawOnClick`

File: frontend/packages/web/src/components/features/DrawButton/DrawButton.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        ...game,
+        ...gameResponse,
+      });
+    });
+  };
+
+  function getDrawState() {
+    const drawStates = game.approved_draw_players;
+
+    if (!drawStates) return 'game_not_started';
```
Comment (izakfr) [inline frontend/packages/web/src/components/features/DrawButton/DrawButton.tsx:30]: Can you make an enum for draw state?
States would be:
```
NONE
INCOMING_DRAW
OUTGOING_DRAW
```

File: frontend/packages/web/src/components/features/DrawButton/DrawButton.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        return 'incoming_draw';
+      }
+    }
+    return 'no_pending_draw';
+  }
+
+  function getDrawProgress() {
+    let num_approved = 0;
+    let num_total = 0;
+    for (const id in game.approved_draw_players) {
```
Comment (izakfr) [inline frontend/packages/web/src/components/features/DrawButton/DrawButton.tsx:44]: you can solve this with an map call:
```
let approvals = Object.values(game.approved_draw_players)
return approvals.filter(val => val).length / approvals.length
```

File: frontend/packages/web/src/components/pages/hompage-service.ts [hunk 4]
```diff
@@ -109,6 +110,20 @@ export class HomepageService {
     });
   }
 
+  drawGame(accepted: boolean, gameID: string, playerID: string) {
```
Comment (izakfr) [inline frontend/packages/web/src/components/pages/hompage-service.ts:113]: can you change the function parameter  to `approved` from `accepted`

**Review by izakfr — CHANGES_REQUESTED** [review]

File: frontend/packages/web/src/components/features/DrawButton/DrawButton.tsx (hunk 3 above)
Comment (jbsummoner) [inline frontend/packages/web/src/components/features/DrawButton/DrawButton.tsx:44]: we prob couldve named this(game.approved_draw_players) @izakfr

**Review by izakfr — APPROVED** [review]
Looks good, thanks!

**Pull request merged**